src/main/java/se/sundsvall/chat/util/ServiceUtil.java

```diff
@@ -0,0 +1,32 @@
+package se.sundsvall.chat.util;
+
+import static org.springframework.util.MimeTypeUtils.APPLICATION_OCTET_STREAM_VALUE;
+
+import java.io.InputStream;
+import org.apache.tika.Tika;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ServiceUtil {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(ServiceUtil.class);
+	private static final String MIME_ERROR_MSG = "Exception when detecting mime type of file with filename '%s'";
+	private static final Tika DETECTOR = new Tika();
+
+	private ServiceUtil() {}
+
+	private static String sanitizeFileName(String fileName) {
+		// Allow only alphanumeric characters, underscores, hyphens, and dots
+		return fileName == null ? "unknown" : fileName.replaceAll("[^a-zA-Z0-9._-]", "_");
+	}
+
+	public static String detectMimeTypeFromStream(final String fileName, final InputStream stream) {
+		try {
```

Comment:
This should be a util in dept44 i.m.h.o.

Reply:
I agree